Null list elements cause cryptic error message

Example pom.xml:

```
...
<configuration>
  <args>
    <arg></arg>
  </args>
</configuration>
...
```

Causes this error when running `mvn compile jib:buildDocker -X`:

```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild (default-cli) on project hello-maven: Execution default-cli of goal com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild failed: at index 0 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild (default-cli) on project hello-maven: Execution default-cli of goal com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild failed: at index 0
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:993)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:345)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:191)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal com.google.cloud.tools:jib-maven-plugin:0.9.0:dockerBuild failed: at index 0
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:145)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:993)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:345)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:191)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException: at index 0
    at com.google.common.collect.ObjectArrays.checkElementNotNull (ObjectArrays.java:235)
    at com.google.common.collect.ObjectArrays.checkElementsNotNull (ObjectArrays.java:225)
    at com.google.common.collect.ObjectArrays.checkElementsNotNull (ObjectArrays.java:219)
    at com.google.common.collect.ImmutableList.construct (ImmutableList.java:348)
    at com.google.common.collect.ImmutableList.copyOf (ImmutableList.java:258)
    at com.google.cloud.tools.jib.builder.BuildConfiguration$Builder.build (BuildConfiguration.java:154)
    at com.google.cloud.tools.jib.maven.BuildDockerMojo.execute (BuildDockerMojo.java:86)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:993)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:345)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:191)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```

Same problem occurs with jvm flags. Empty lists (e.g. `<args></args>`) do not cause the build to fail.

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ContainerConfiguration.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ContainerConfiguration.java
@@ -72,7 +72,8 @@ public class ContainerConfiguration {
       if (programArguments == null) {
         this.programArguments = null;
       } else {
-        Preconditions.checkArgument(!programArguments.contains(null));
+        Preconditions.checkArgument(
+            !programArguments.contains(null), "args list contains null elements");
         this.programArguments = ImmutableList.copyOf(programArguments);
       }
       return this;
@@ -88,8 +89,12 @@ public class ContainerConfiguration {
       if (environmentMap == null) {
         this.environmentMap = null;
       } else {
-        Preconditions.checkArgument(!Iterables.any(environmentMap.keySet(), Objects::isNull));
-        Preconditions.checkArgument(!Iterables.any(environmentMap.values(), Objects::isNull));
+        Preconditions.checkArgument(
+            !Iterables.any(environmentMap.keySet(), Objects::isNull),
+            "environment map contains null keys");
+        Preconditions.checkArgument(
+            !Iterables.any(environmentMap.values(), Objects::isNull),
+            "environment map contains null values");
         this.environmentMap = new HashMap<>(environmentMap);
       }
       return this;
@@ -112,7 +117,8 @@ public class ContainerConfiguration {
       if (exposedPorts == null) {
         this.exposedPorts = null;
       } else {
-        Preconditions.checkArgument(!exposedPorts.contains(null));
+        Preconditions.checkArgument(
+            !exposedPorts.contains(null), "ports list contains null elements");
         this.exposedPorts = new HashSet<>(exposedPorts);
       }
       return this;
@@ -135,7 +141,7 @@ public class ContainerConfiguration {
       if (volumes == null) {
         this.volumes = null;
       } else {
-        Preconditions.checkArgument(!volumes.contains(null));
+        Preconditions.checkArgument(!volumes.contains(null), "volumes list contains null elements");
         this.volumes = new HashSet<>(volumes);
       }
       return this;
@@ -158,8 +164,10 @@ public class ContainerConfiguration {
       if (labels == null) {
         this.labels = null;
       } else {
-        Preconditions.checkArgument(!Iterables.any(labels.keySet(), Objects::isNull));
-        Preconditions.checkArgument(!Iterables.any(labels.values(), Objects::isNull));
+        Preconditions.checkArgument(
+            !Iterables.any(labels.keySet(), Objects::isNull), "labels map contains null keys");
+        Preconditions.checkArgument(
+            !Iterables.any(labels.values(), Objects::isNull), "labels map contains null values");
         this.labels = new HashMap<>(labels);
       }
       return this;
@@ -182,7 +190,8 @@ public class ContainerConfiguration {
       if (entrypoint == null) {
         this.entrypoint = null;
       } else {
-        Preconditions.checkArgument(!entrypoint.contains(null));
+        Preconditions.checkArgument(
+            !entrypoint.contains(null), "entrypoint contains null elements");
         this.entrypoint = ImmutableList.copyOf(entrypoint);
       }
       return this;

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ContainerConfiguration.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ContainerConfiguration.java
@@ -73,7 +73,7 @@ public class ContainerConfiguration {
         this.programArguments = null;
       } else {
         Preconditions.checkArgument(
-            !programArguments.contains(null), "args list contains null elements");
+            !programArguments.contains(null), "program arguments list contains null elements");
         this.programArguments = ImmutableList.copyOf(programArguments);
       }
       return this;

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ImageConfiguration.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/configuration/ImageConfiguration.java
@@ -39,7 +39,8 @@ public class ImageConfiguration {
      * @return this
      */
     public Builder setCredentialRetrievers(List<CredentialRetriever> credentialRetrievers) {
-      Preconditions.checkArgument(!credentialRetrievers.contains(null));
+      Preconditions.checkArgument(
+          !credentialRetrievers.contains(null), "credential retriever list contains null elements");
       this.credentialRetrievers = ImmutableList.copyOf(credentialRetrievers);
       return this;
     }

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/configuration/ContainerConfigurationTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/configuration/ContainerConfigurationTest.java
@@ -38,7 +38,7 @@ public class ContainerConfigurationTest {
       ContainerConfiguration.builder().setProgramArguments(Arrays.asList("first", null));
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("args list contains null elements", ex.getMessage());
     }
 
     // Entrypoint element should not be null.
@@ -46,7 +46,7 @@ public class ContainerConfigurationTest {
       ContainerConfiguration.builder().setEntrypoint(Arrays.asList("first", null));
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("entrypoint contains null elements", ex.getMessage());
     }
 
     // Exposed ports element should not be null.
@@ -55,38 +55,54 @@ public class ContainerConfigurationTest {
       ContainerConfiguration.builder().setExposedPorts(badPorts);
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("ports list contains null elements", ex.getMessage());
     }
 
-    // Labels element should not be null.
-    Map<String, String> badLabels = new HashMap<>();
-    badLabels.put("label-key", null);
+    // Volume element should not be null.
+    Set<AbsoluteUnixPath> badVolumes =
+        new HashSet<>(Arrays.asList(AbsoluteUnixPath.get("/"), null));
     try {
-      ContainerConfiguration.builder().setLabels(badLabels);
+      ContainerConfiguration.builder().setVolumes(badVolumes);
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("volumes list contains null elements", ex.getMessage());
     }
 
-    // Environment keys element should not be null.
     Map<String, String> nullKeyMap = new HashMap<>();
     nullKeyMap.put(null, "value");
+    Map<String, String> nullValueMap = new HashMap<>();
+    nullValueMap.put("key", null);
 
+    // Label keys should not be null.
+    try {
+      ContainerConfiguration.builder().setLabels(nullKeyMap);
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertEquals("labels map contains null keys", ex.getMessage());
+    }
+
+    // Labels values should not be null.
+    try {
+      ContainerConfiguration.builder().setLabels(nullValueMap);
+      Assert.fail("The IllegalArgumentException should be thrown.");
+    } catch (IllegalArgumentException ex) {
+      Assert.assertEquals("labels map contains null values", ex.getMessage());
+    }
+
+    // Environment keys should not be null.
     try {
       ContainerConfiguration.builder().setEnvironment(nullKeyMap);
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("environment map contains null keys", ex.getMessage());
     }
 
-    // Environment values element should not be null.
-    Map<String, String> nullValueMap = new HashMap<>();
-    nullValueMap.put("key", null);
+    // Environment values should not be null.
     try {
       ContainerConfiguration.builder().setEnvironment(nullValueMap);
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
+      Assert.assertEquals("environment map contains null values", ex.getMessage());
     }
   }
 

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/configuration/ContainerConfigurationTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/configuration/ContainerConfigurationTest.java
@@ -38,7 +38,7 @@ public class ContainerConfigurationTest {
       ContainerConfiguration.builder().setProgramArguments(Arrays.asList("first", null));
       Assert.fail("The IllegalArgumentException should be thrown.");
     } catch (IllegalArgumentException ex) {
-      Assert.assertEquals("args list contains null elements", ex.getMessage());
+      Assert.assertEquals("program arguments list contains null elements", ex.getMessage());
     }
 
     // Entrypoint element should not be null.

